src/fides/api/ctl/migrations/versions/643249f65453_case_insensitive_usernames_delete_.py

```diff
@@ -0,0 +1,48 @@
+"""case_insensitive_usernames_delete_duplicates
+
+Revision ID: 643249f65453
+Revises: 392992c7733a
+Create Date: 2023-02-01 22:22:38.055862
+
+"""
+from alembic import op
+from sqlalchemy.orm.session import Session
+from sqlalchemy.sql import text
+
+from fides.lib.models.fides_user import FidesUser
+from fides.lib.models.fides_user_permissions import FidesUserPermissions
+
+# revision identifiers, used by Alembic.
+revision = "643249f65453"
+down_revision = "392992c7733a"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+
+    # Found here in the 24.2.2.4. Nondeterministic Collations section:
+    # https://www.postgresql.org/docs/current/collation.html
+    op.execute(
+        "CREATE COLLATION IF NOT EXISTS case_insensitive (provider = icu, locale = 'und-u-ks-level2', deterministic = false);"
+    )
+    session = Session(bind=op.get_bind())
+    conn = session.connection()
+    statement = text(
+        """SELECT DISTINCT ON (username) username collate case_insensitive, created_at, id
+FROM fidesuser
+ORDER BY username, created_at ASC"""
+    )
+
+    result = conn.execute(statement)
+    original_user_ids = [row[2] for row in result]
+    session.query(FidesUserPermissions).filter(
+        FidesUserPermissions.user_id.not_in(original_user_ids)
+    ).delete()
```

Comment:
Also look into deleting associated user_ids from the `client` table - that will prevent a migration.  But I wouldn't delete the `client` record itself, since several other resources reference it.

Reply:
@pattisdr I deleted those records because I was running into foreign key constraint errors when deleting the users. It turns out that the `user_id` column on `FidesUserPermissions` can't be null

https://github.com/ethyca/fides/blob/56b888e39e0fd013860aec0bb492ccfb994aac28/src/fides/lib/models/fides_user_permissions.py#L15

We would need to update the model and maybe a migration to allow that field to be nullable.

Does it make sense to delete the permission records with that in mind?

Reply:
I am talking about a separate model Andrew, I'm fine with what you did with the FidesUserPermissions records, that makes sense.

I was talking about the separate `Client` model that has a FK to `fidesuser`  I think you need to remove associated user_ids. That column is already nullable, so I'd just remove the data.

Reply:
oh and it'd also be good to double check there are no other tables referencing user_id

Reply:
I understand now. I'll look into that now